Request/ParamConverter/PsrServerRequestParamConverter.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sensio\Bundle\FrameworkExtraBundle\Request\ParamConverter;
+
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\ParamConverter;
+use Symfony\Bridge\PsrHttpMessage\Factory\DiactorosFactory;
+use Symfony\Bridge\PsrHttpMessage\HttpMessageFactoryInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Converts HttpFoundation Request to PSR-7 ServerRequest using the bridge.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+class PsrServerRequestParamConverter implements ParamConverterInterface
+{
+    private static $supportedTypes = [
+        'Psr\Http\Message\ServerRequestInterface' => true,
+        'Psr\Http\Message\RequestInterface' => true,
+        'Psr\Http\Message\HttpMessageInterface' => true,
+    ];
+    /**
+     * @var HttpMessageFactoryInterface
+     */
+    private $httpMessageFactory;
+
+    public function __construct(HttpMessageFactoryInterface $httpMessageFactory = null)
+    {
+        if (null === $httpMessageFactory) {
+            if (!class_exists('Symfony\Bridge\PsrHttpMessage\Factory\DiactorosFactory')) {
+                throw new \RuntimeException('Install the PSR-7 bridge to inject a Psr\Http\Message\ServerRequestInterface instance in the controller.');
+            }
+
+            $httpMessageFactory = new DiactorosFactory();
+        }
+
+        $this->httpMessageFactory = $httpMessageFactory;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function apply(Request $request, ParamConverter $configuration)
+    {
+        $request->attributes->set($configuration->getName(), $this->httpMessageFactory->createRequest($request));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supports(ParamConverter $configuration)
+    {
+        if (null === $configuration->getClass()) {
+            return false;
+        }
+
+        return isset(self::$supportedTypes[$configuration->getClass()]);
```

Comment:
As variable is private, we can style update our CS once merged to allow or disallow that kind of tricks and change it here if necessary.